src/AutoRest.CSharp/Common/Output/Models/Types/ModelFactoryTypeProvider.cs

```diff
@@ -26,25 +26,34 @@ internal sealed class ModelFactoryTypeProvider : TypeProvider
         protected override string DefaultName { get; }
         protected override string DefaultAccessibility { get; }
 
-        private IEnumerable<MethodSignature>? _methods;
-        public IEnumerable<MethodSignature> Methods => _methods ??= Models.Select(CreateMethod);
+        private IReadOnlyList<MethodSignature>? _methods;
+        public IReadOnlyList<MethodSignature> Methods => _methods ??= Models!.Select(CreateMethod).ToList();
 
-        public IEnumerable<SerializableObjectType> Models { get; }
+        public IEnumerable<SerializableObjectType>? Models { get; }
```

Comment:
it is not really good to make a collection nullable because an empty list of models look like means just the same as null.

I saw we have a new ctor for this type provider, if the ctor must initialize this property, we could initialize it as `Enumerable.Empty<SerializableObjectType>()`

Reply:
The new constructor is removed, this can be reverted.